### ISSUE TITLE
[accessibility][voice reader] When using talkback on Android during forward navigation focus moves out from a drop down list

### ISSUE THREAD

**Issue by lutorre** [microsoft/fluentui#10637]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)
Google chrome on Android (android 9 pie)

> Note that our screen reader support is limited to Edge + Narrator. Issues that do not reproduce in Edge + Narrator, and aren't caused by obvious invalid aria values, should be filed with the respective screen reading software, not the Fabric repo.

### Describe the issue:

On Android devices focus moves to elements outside a drop down list while navigating after last item in the list in forward direction

#### Actual behavior:

Focus moves to elements outside the drop down list while navigating after last item in the list in forward direction.

#### Expected behavior:
Focus should move to the first item in the list when user navigates in forward direction after the last item.

**Comment by joschect** [comment]
@lutorre I have been able to reproduce this using swipe right to navigate through the available items. We'll add it to our backlog.

**Comment by JustSlone** [comment]
CodePen with Repro: https://codepen.io/justslone-the-encoder/pen/GRjqQdJ?editors=0010

Repro: Turn on scan mode of any Screen reader and navigate with right arrow, you'll eventually leave the date picker dialog. 
(sorry for the crummy gif, recorder was giving me trouble)
![DatePicker-Narrator](https://user-images.githubusercontent.com/40304143/101444583-a658ed00-38d4-11eb-9ab9-7d61b16068d0.gif)

Since DatePicker dialog is not a modal dialog and therefore does not trap the screen reader. 

This seems like it might be by design? 

For comparison we can see that this behavior repros on the Aria 1.1 Date Picker example https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/datepicker-dialog.html

**Comment by smhigley** [comment]
I agree that it's by design, dropdowns for comboboxes generally aren't modal. That might be an interesting study question sometime later if screen reader users complain about the behavior, but for now it makes sense to follow the standard non-modal pattern IMO.

**Comment by JustSlone** [comment]
Thanks @smhigley 

On that note, I'm going to close this issue. @lutorre (or anyone else) feel free to reopen if there is new information or something was missed.
